flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/benchmark/StreamNetworkBenchmarkEnvironment.java

```diff
@@ -249,18 +250,21 @@ private InputGate createInputGate(
 				channel,
 				channelDescriptors);
 
-			SingleInputGate gate = new SingleInputGateFactory(
+			final SingleInputGateFactory gateFactory = new SingleInputGateFactory(
 				environment.getConfiguration(),
 				environment.getConnectionManager(),
 				environment.getResultPartitionManager(),
 				new TaskEventDispatcher(),
-				environment.getNetworkBufferPool())
-				.create(
+				environment.getNetworkBufferPool());
+
+			final InputGate gate = new InputGateWithMetrics(
+				gateFactory.create(
```

Comment:
nit: maybe have a separate variable for `gateFactory.create`
also channel loop body could be refactored into two methods: `createInputGateDeploymentDescriptor` and `createSingleInputGateWithMetrics` to reduce the current method

Reply:
Actually gate factory should not be created inside loop, and it could be created only once together with `NetworkEnvironment`. I would submit a hotfix commit firstly for refactoring this method, and then append my PR based on it.